### PR TITLE
feat: inject secrets via module hook

### PR DESCRIPTION
**Pull request by george-e-shaw-iv**
* adds the ability to inject vault secrets via a module hook

### CODE REVIEW

File: templates/deployments/appname/app.config.jsonnet.tpl
Comment (george-e-shaw-iv) [inline templates/deployments/appname/app.config.jsonnet.tpl:1]: The changes in this file actually allow config mixins to work. They weren't possible before the additions of the `+` operator (idk what it is called in jsonnet).

**Review by malept — CHANGES_REQUESTED** [review]
The changes themselves seem fine. Once you've updated the test snapshots, can you add some docs to the [module hooks markdown file](https://github.com/getoutreach/stencil-golang/blob/v1.23.3/docs/module-hooks.md)?

File: docs/module-hooks.md [hunk 1]
```diff
@@ -36,6 +36,18 @@ Equivalent `go_modules` but for JavaScript (node).
 
 Equivalent `go_modules` but for JavaScript (node), dev dependencies.
 
+### `vaultSecrets`
+
+**Type**: `string`
+
+**File**: `_helpers.tpl`
```
Comment (malept) [inline docs/module-hooks.md:43]: suggestion: I think ultimately this should point to the jsonnet template.
```suggestion
**File**: `deployments/appname/app.jsonnet.tpl`
```

File: docs/module-hooks.md [hunk 2]
```diff
@@ -36,6 +36,18 @@ Equivalent `go_modules` but for JavaScript (node).
 
 Equivalent `go_modules` but for JavaScript (node), dev dependencies.
 
+### `vaultSecrets`
+
+**Type**: `string`
```
Comment (malept) [inline docs/module-hooks.md:41]: suggestion: list of strings?
```suggestion
**Type**: `[]string`
```

File: docs/module-hooks.md [hunk 3]
```diff
@@ -36,6 +36,18 @@ Equivalent `go_modules` but for JavaScript (node).
...
+
+**File**: `_helpers.tpl`
+
+Adds a vault secret to be pulled within the deployment manifests and created in kubernetes.
```
Comment (malept) [inline docs/module-hooks.md:45]: Nit:
```suggestion
Adds Vault secret paths to be pulled within the deployment manifests and created in Kubernetes.
```

**Review by malept — APPROVED** [review]

**Pull request merged**